socs/agents/synacc/agent.py

```diff
@@ -17,14 +17,19 @@ class SynaccessAgent:
         password(str): Password credential to login to device.
     """
 
-    def __init__(self, agent, ip_address, username, password):
+    def __init__(self, agent, ip_address, username, password, outlet_names=None):
         self.agent = agent
         self.log = agent.log
         self.lock = TimeoutLock()
         self.ip_address = ip_address
         self.user = username
         self.passw = password
 
+        if outlet_names is None:
+            self.outlet_names = ['outlet1', 'outlet2', 'outlet3', 'outlet4', 'outlet5']
+        else:
+            self.outlet_names = outlet_names
```

Comment:
Should we check the length of `outlet_names` here?  I guess it's pretty easy to debug an indexing error later.

But I think it might be nice to pad the list, so a user only has to declare the You could even make it "pad" the list with any missing entries;
```
for i in range(MAX_POSSIBLE_OUTLETS):
  if len(outlet_names) <= i:
    outlet_names.append('outlet%i' % (i+1))
```